tokio/src/runtime/builder.rs

```diff
@@ -374,6 +382,79 @@ impl Builder {
         self
     }
 
+    /// Executes function `f` just before a thread is parked (goes idle).
+    /// `f` is called within the Tokio context, so functions like [`tokio::spawn`](crate::spawn)
+    /// can be called, and may result in this thread being unparked immediately.
+    ///
+    /// This can be used to start work only when the executor is idle, or for bookkeeping
+    /// and monitoring purposes.
+    ///
```

Comment:
Should the docs maybe state explicitly that calling this multiple times will overwrite any previously-registered callbacks, rather than registering multiple callbacks? That seems like a potential footgun, if a user calls `on_thread_park` multiple times...